cmd/kubeadm/app/apis/output/v1alpha3/types.go

```diff
@@ -73,12 +73,23 @@ type ComponentConfigVersionState struct {
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
+// AvailableUpgrade represents information for a single avaliable upgrade.
+type AvailableUpgrade struct {
+	metav1.TypeMeta
+
+	Description string `json:"description"`
+
+	Components []ComponentUpgradePlan `json:"components"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
 // UpgradePlan represents information about upgrade plan for the output
 // produced by 'kubeadm upgrade plan'
 type UpgradePlan struct {
 	metav1.TypeMeta
 
-	Components []ComponentUpgradePlan `json:"components"`
+	AvailableUpgrades []AvailableUpgrade `json:"availableUpgrades"`
 
 	ConfigVersions []ComponentConfigVersionState `json:"configVersions"`
```

Comment:
we can make this `json:"configVersions,omitempty"` to avoid showing it as `null`